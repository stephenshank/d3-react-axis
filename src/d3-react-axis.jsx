import React from "react";

function D3ReactAxis() {
  return <h2>Axis</h2>;
}

export default D3ReactAxis;
