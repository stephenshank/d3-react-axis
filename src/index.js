import D3ReactAxis from "./d3-react-axis.jsx";

export default D3ReactAxis;
