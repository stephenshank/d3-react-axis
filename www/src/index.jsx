import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter, Route, NavLink } from "react-router-dom";
import "bootstrap";
import { scaleLinear } from "d3-scale";
import { AxisTop, AxisRight, AxisBottom, AxisLeft } from "d3-react-axis";

import "bootstrap/dist/css/bootstrap.min.css";

function Link(props) {
  return (
    <NavLink className="dropdown-item link" to={props.to}>
      {props.header}
    </NavLink>
  );
}

function Dropdown(props) {
  return (
    <ul className="navbar-nav ">
      <li className="nav-item dropdown">
        <a
          className="nav-link dropdown-toggle"
          href="#"
          id="navbarDropdown"
          role="button"
          data-toggle="dropdown"
          aria-haspopup="true"
          aria-expanded="false"
        >
          {props.title}
        </a>
        <div className="dropdown-menu" aria-labelledby="navbarDropdown">
          {props.children}
        </div>
      </li>
    </ul>
  );
}

function NavBar() {
  return (<nav className="navbar navbar-expand-lg navbar-light bg-light">
    <NavLink className="navbar-brand" to="/">
      D3-React-Axis
    </NavLink>
    <Dropdown title="Examples">
      <Link to="/mwe" header="Minimal working example" />
    </Dropdown>
  </nav>);
}

function Home() {
  return <h1>Home</h1>;
}

function MWE() {
  const long_side = 400,
    short_side = 60,
    padding = 30,
    scale = scaleLinear()
      .domain([0, 100])
      .range([0, long_side-2*padding]),
    border_style = {border: "solid", borderWidth: 1, borderColor: "LightGrey"};

  return (<div>
    <h1>Minimal working examples</h1>
    <h2>AxisTop</h2>
    <svg
      width={long_side}
      height={short_side}
      style={border_style}
    >
      <AxisTop
        scale={scale}
        transform={`translate(${padding}, ${short_side-padding})`}
      />
    </svg>

    <h2>AxisBottom</h2>
    <svg
      width={long_side}
      height={short_side}
      style={border_style}
    >
      <AxisBottom
        scale={scale}
        transform={`translate(${padding}, ${padding})`}
      />
    </svg>

    <h2>AxisLeft</h2>
    <svg
      width={short_side}
      height={long_side}
      style={border_style}
    >
      <AxisLeft
        scale={scale}
        transform={`translate(${padding}, ${padding})`}
      />
    </svg>

    <h2>AxisRight</h2>
    <svg
      width={short_side}
      height={long_side}
      style={border_style}
    >
      <AxisRight
        scale={scale}
        transform={`translate(${padding}, ${padding})`}
      />
    </svg>
  </div>);
}

ReactDOM.render(
  <BrowserRouter>
    <div>
      <NavBar />
      <div style={{ maxWidth: 1140 }} className="container-fluid">
        <Route exact path="/" component={Home} />
        <Route path="/mwe" component={MWE} />
      </div>
    </div>
  </BrowserRouter>,
  document.body.appendChild(document.createElement("div"))
);

