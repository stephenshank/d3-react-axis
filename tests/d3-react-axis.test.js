import React from "react";
import { scaleLinear } from "d3-scale";
import renderer from "react-test-renderer";

import { AxisTop, AxisRight, AxisBottom, AxisLeft } from "../src/index";


const width = 500,
  height = 50;

function SVG(props) {
  return (<svg width={width} height={height}>
    {props.children}
  </svg>);
}

const domain = [0, 100],
  range = [0, 400];

it("axis top with linear scale", () => {
  const scale = scaleLinear()
      .domain(domain)
      .range(range),
    axis_top = renderer
      .create(<SVG>
        <AxisTop
          scale={scale}
        />
      </SVG>)
      .toJSON();
  expect(axis_top).toMatchSnapshot();
});

it("axis right with linear scale", () => {
  const scale = scaleLinear()
      .domain(domain)
      .range(range),
    axis_top = renderer
      .create(<SVG>
        <AxisRight
          scale={scale}
        />
      </SVG>)
      .toJSON();
  expect(axis_top).toMatchSnapshot();
});

it("axis bottom with linear scale", () => {
  const scale = scaleLinear()
      .domain(domain)
      .range(range),
    axis_top = renderer
      .create(<SVG>
        <AxisBottom
          scale={scale}
        />
      </SVG>)
      .toJSON();
  expect(axis_top).toMatchSnapshot();
});

it("axis left with linear scale", () => {
  const scale = scaleLinear()
      .domain(domain)
      .range(range),
    axis_top = renderer
      .create(<SVG>
        <AxisLeft
          scale={scale}
        />
      </SVG>)
      .toJSON();
  expect(axis_top).toMatchSnapshot();
});
