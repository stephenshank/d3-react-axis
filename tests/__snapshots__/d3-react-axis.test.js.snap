// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axis bottom with linear scale 1`] = `
<svg
  height={50}
  width={500}
>
  <g
    fill="none"
    fontFamily="sans-serif"
    fontSize={10}
    textAnchor="middle"
    transform="translate(0, 0)"
  >
    <path
      className="domain"
      d="M0.5,6V0.5H400.5V6"
      stroke="currentColor"
    />
    <g
      className="tick"
      transform="translate(0.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        0
      </text>
    </g>
    <g
      className="tick"
      transform="translate(40.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        10
      </text>
    </g>
    <g
      className="tick"
      transform="translate(80.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        20
      </text>
    </g>
    <g
      className="tick"
      transform="translate(120.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        30
      </text>
    </g>
    <g
      className="tick"
      transform="translate(160.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        40
      </text>
    </g>
    <g
      className="tick"
      transform="translate(200.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        50
      </text>
    </g>
    <g
      className="tick"
      transform="translate(240.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        60
      </text>
    </g>
    <g
      className="tick"
      transform="translate(280.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        70
      </text>
    </g>
    <g
      className="tick"
      transform="translate(320.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        80
      </text>
    </g>
    <g
      className="tick"
      transform="translate(360.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        90
      </text>
    </g>
    <g
      className="tick"
      transform="translate(400.5,0)"
    >
      <line
        stroke="currentColor"
        y2={6}
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y={9}
      >
        100
      </text>
    </g>
  </g>
</svg>
`;

exports[`axis left with linear scale 1`] = `
<svg
  height={50}
  width={500}
>
  <g
    fill="none"
    fontFamily="sans-serif"
    fontSize={10}
    textAnchor="end"
    transform="translate(0, 0)"
  >
    <path
      className="domain"
      d="M-6,0.5H0.5V400.5H-6"
      stroke="currentColor"
    />
    <g
      className="tick"
      transform="translate(0,0.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        0
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,40.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        10
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,80.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        20
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,120.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        30
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,160.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        40
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,200.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        50
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,240.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        60
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,280.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        70
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,320.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        80
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,360.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        90
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,400.5)"
    >
      <line
        stroke="currentColor"
        x2={-6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={-9}
      >
        100
      </text>
    </g>
  </g>
</svg>
`;

exports[`axis right with linear scale 1`] = `
<svg
  height={50}
  width={500}
>
  <g
    fill="none"
    fontFamily="sans-serif"
    fontSize={10}
    textAnchor="start"
    transform="translate(0, 0)"
  >
    <path
      className="domain"
      d="M6,0.5H0.5V400.5H6"
      stroke="currentColor"
    />
    <g
      className="tick"
      transform="translate(0,0.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        0
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,40.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        10
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,80.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        20
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,120.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        30
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,160.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        40
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,200.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        50
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,240.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        60
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,280.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        70
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,320.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        80
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,360.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        90
      </text>
    </g>
    <g
      className="tick"
      transform="translate(0,400.5)"
    >
      <line
        stroke="currentColor"
        x2={6}
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x={9}
      >
        100
      </text>
    </g>
  </g>
</svg>
`;

exports[`axis top with linear scale 1`] = `
<svg
  height={50}
  width={500}
>
  <g
    fill="none"
    fontFamily="sans-serif"
    fontSize={10}
    textAnchor="middle"
    transform="translate(0, 0)"
  >
    <path
      className="domain"
      d="M0.5,-6V0.5H400.5V-6"
      stroke="currentColor"
    />
    <g
      className="tick"
      transform="translate(0.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        0
      </text>
    </g>
    <g
      className="tick"
      transform="translate(40.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        10
      </text>
    </g>
    <g
      className="tick"
      transform="translate(80.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        20
      </text>
    </g>
    <g
      className="tick"
      transform="translate(120.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        30
      </text>
    </g>
    <g
      className="tick"
      transform="translate(160.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        40
      </text>
    </g>
    <g
      className="tick"
      transform="translate(200.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        50
      </text>
    </g>
    <g
      className="tick"
      transform="translate(240.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        60
      </text>
    </g>
    <g
      className="tick"
      transform="translate(280.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        70
      </text>
    </g>
    <g
      className="tick"
      transform="translate(320.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        80
      </text>
    </g>
    <g
      className="tick"
      transform="translate(360.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        90
      </text>
    </g>
    <g
      className="tick"
      transform="translate(400.5,0)"
    >
      <line
        stroke="currentColor"
        y2={-6}
      />
      <text
        dy="0em"
        fill="currentColor"
        y={-9}
      >
        100
      </text>
    </g>
  </g>
</svg>
`;
